pkg/util/fs_util.go

```diff
@@ -71,25 +71,58 @@ var excluded []string
 
 type ExtractFunction func(string, *tar.Header, io.Reader) error
 
+type FSConfig struct {
+	includeWhiteout bool
+	extractFunc     ExtractFunction
+}
+
+type FSOpt func(*FSConfig)
+
+func IncludeWhiteout() FSOpt {
+	return func(opts *FSConfig) {
+		opts.includeWhiteout = true
```

Comment:
this is always true. Do you see any future this being configured via command line? i am wondering if we could make the code simpler and avoid this config.

Reply:
I don't see it ever being configured via command line since a single process will call FSFromLayers with different options.

This is always true because the zero value is false. So by not including this option it is set to false.

I might be misunderstanding

Reply:
I guess I could see an argument for passing the value of IncludeWhitespace as an arg to the function so it could be switched from true to false but that seems like something that could be changed later if the need arisea

Reply:
actually i meant the opposite. Lets remove `opts.includeWhiteout` totally

Reply:
I don't understand. When `GetFSFromImage` calls `GetFSFromLayers` `includeWhiteout` should be false. When `CachingCopyCommand` or `CachingRunCommand` calls `GetFSFromLayers` `includeWhiteout` should be true.

so `includeWhiteout` has to be configurable in some regards. Are you objecting to having it be configurable or this specific configuration pattern?

Reply:
Ah, i think it did not occur to me `includeWhiteout` was false due to the `opts .. FSOpt` array. 
That explains.

```
	return GetFSFromLayers(root, layers, ExtractFunc(extract))
}

func GetFSFromLayers(root string, layers []v1.Layer, opts ...FSOpt) ([]string, error) {

```